src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -491,9 +497,83 @@ public static String getIndexName(String cfName)
         return StringUtils.substringAfter(cfName, Directories.SECONDARY_INDEX_NAME_SEPARATOR);
     }
 
+    // TODO: JavaDoc if this approach survives...mention that this should not be used for live SSTables
+    public void validateStorageAttachedIndexes(Collection<SSTableReader> sstables) throws IOException
+    {
+        // TODO: A more generic approach via the Index interface rather than just working w/ SAI 
```

Comment:
Indeed the entire method body seems SAI-specific, so I guess it should be in the Index interface.

Reply:
Perhaps this is something I can push down to the `Group` interface, then have implemented by SAI and no-op'd by everything else for now...or validate only for SSTable-attached things. On `SIM` the method could be something like `validateSSTableAttachedIndexes()`? (Still need to access the `Group` somehow, so I guess `SIM` is the facade.)

Reply:
Probably SIM should call a validation method in each `Group`, and the `Group` would call another validation method on each of its indexes. That way the method on SIM doesn't need to know anything about `StorageAttachedIndex`, and can have a more generic name such as `validateStreamedIndexes` or something like that?

Reply:
This is basically what I ended up doing.